src/libaudtag/id3/id3v24.cc

```diff
@@ -664,6 +697,20 @@ bool ID3v24TagModule::write_tag (VFSFile & f, const Tuple & tuple)
     String comment = tuple.get_str (Tuple::Comment);
     add_comment_frame (comment, dict);
 
+    String lyrics = tuple.get_str (Tuple::Lyrics);
+    if (lyrics && lyrics[9])
+    {
+        add_lyrics_frame (lyrics, dict);
+        mustpreservelength = true;
+    }
+
+    if (mustpreservelength)
+    {
+        int decoder_length = tuple.get_int (Tuple::Length);
+        if (decoder_length > 0)
+            add_frameFromTupleInt (tuple, Tuple::Length, ID3_LENGTH, dict);
```

Comment:
This is wrong. The `Tuple::Length` value usually comes from what the decoder calculated. We shouldn't be writing that value back to the ID3 tag.

Reply:
Ok, I removed this. I didn't know before how this feature works and how the value is calculated. I haven't tested this part of code yet.